ecpy/utils/atom_util.py

```diff
@@ -53,41 +65,158 @@ def tagged_members(obj, meta=None, meta_value=None):
                 member.metadata[meta] == meta_value}
 
 
-def member_from_str(member, str_value):
-    """ Convert a string to the right type for a member.
+def member_from_pref(obj, member, val):
+    """ Retrieve the value stored in the preferences for a member.
 
     Parameters
     ----------
+    obj : Atom
+        Object who owns the member.
+
     member : Member
-        Member for which the string should be converted to a value
+        Member for which the preferences should be retrieved.
 
-    str_value : string
-        String to convert
+    val : Value
+        Value that is stored in the preferences, depending on the case this
+        might be a serialized value or simply a string.
 
     Returns
     -------
-    converted_value
-        The converted value
+    value
+        The Deserialized value that can be assigned to the member.
 
     """
-    # If the member type is Str or Unicode then we just take the raw value and
-    # Atom will handle the casting if any for us.
-    if isinstance(member, (Str, Unicode)):
-        value = str_value
+    meta_value = member.metadata[PREF_KEY]
+
+    # If 'pref=True' then we rely on the standard save mechanism
+    if meta_value is True:
+        # If the member is a subclass of Str, Unicode then we just take the
+        # raw value and Atom will handle the casting if any for us.
+        # If it is a subclass of basestring then we save it as-is
+        if isinstance(member, (Str, Unicode, basestring)):
+            value = val
+
+        # If it is an Enum where the first item is a (subclass of) string, then
+        # we assume that the whole Enum contains strings and we save it as-is
+        elif isinstance(member, Enum) and isinstance(member.items[0],
+                                                     basestring):
+            value = val
+
+        # Otherwise, we eval it, or we might throw an error
+        else:
+            try:
+                value = literal_eval(val)
+            except ValueError:
+                # Silently ignore failed evaluation as we can have a string
+                # assigned to a value.
+                value = val
+
+    # If the user provided a custom "from_pref" function, then we check
+    # that it has the correct signature and use it to obtain the value
+    elif (isinstance(meta_value, (Tuple, List, tuple, list)) and
```

Comment:
tuple and list are enough user won't provide a Member (which is a descriptor not a container)